sql/src/main/java/org/apache/druid/sql/calcite/schema/DruidSchema.java

```diff
@@ -431,19 +430,27 @@ void removeSegment(final DataSegment segment)
     }
   }
 
-  private void removeServerSegment(final DruidServerMetadata server, final DataSegment segment)
+  @VisibleForTesting
+  void removeServerSegment(final DruidServerMetadata server, final DataSegment segment)
   {
     synchronized (lock) {
       log.debug("Segment[%s] is gone from server[%s]", segment.getId(), server.getName());
       final Map<SegmentId, AvailableSegmentMetadata> knownSegments = segmentMetadataInfo.get(segment.getDataSource());
       final AvailableSegmentMetadata segmentMetadata = knownSegments.get(segment.getId());
-      final Set<String> segmentServers = segmentMetadata.getReplicas();
-      final ImmutableSet<String> servers = FluentIterable.from(segmentServers)
-                                                         .filter(Predicates.not(Predicates.equalTo(server.getName())))
-                                                         .toSet();
+      final Set<DruidServerMetadata> segmentServers = segmentMetadata.getReplicas();
+      final ImmutableSet<DruidServerMetadata> servers = FluentIterable.from(segmentServers)
+                                                                      .filter(Predicates.not(Predicates.equalTo(server)))
+                                                                      .toSet();
+      final DruidServerMetadata realtimeServer = servers.stream()
+                                                  .filter(metadata -> metadata.getType().equals(ServerType.REALTIME))
+                                                  .findAny()
+                                                  .orElse(null);
```

Comment:
Since `findAny()` returns an `Optional`, you could use `Optional.isPresent()` below instead of using `orElse()` to convert to `null`. Another option is to use `anyMatch()` instead of `findAny()`.

Reply:
ah, sure, i used `Optional.isPresent()`